linter/linter.go

```diff
@@ -34,7 +34,17 @@ func Check(tbs *x509.Certificate, subjectPubKey crypto.PublicKey, realIssuer *x5
 	if err != nil {
 		return err
 	}
-	_, err = linter.Check(tbs, subjectPubKey)
+
+	// If the provided pubkey is identical to the public half of the provided
```

Comment:
This is very cool. And we should document it on the `Check` API.

By the way, is it possible to implement this on `linter.Check`, not just the top-level function? It would be nice if they had consistent behavior.

Reply:
I agree the boolean parameter is inelegant. What about this: `Linter` gains a new field `realSignerPubkey`, stored when it is first created by `New`. We move the check that is currently in the Check top-level function into the Check method on Linter, and have it check "Linter.realSignerPubkey equals subjectPubkey".

Reply:
Ah yeah I like that a lot better, done.